### PR TITLE
Properly handle error code translation

### PR DESCRIPTION
**Pull request by TylerHelmuth**
#### Description

Fixes a bug where the otlpreceiver was not translating grpc error codes to http status codes the same way that the otlp exporter was translating http status codes to grpc error codes.

#### Link to tracking issue
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/10538

#### Testing
Added unit tests

### CODE REVIEW

**Review by evan-bradley — APPROVED** [review]
Verified with [`NewStatusFromMsgAndHTTPCode`](https://github.com/open-telemetry/opentelemetry-collector/blob/af33ac5acc08dbeb43426f86b526d2cfd29318ab/internal/httphelper/helper.go#L15C6-L15C33) and the two conversions should match now.

**Review by jaronoff97 — APPROVED** [review]

File: receiver/otlpreceiver/internal/errors/errors.go [hunk 1]
```diff
@@ -40,6 +40,18 @@ func GetHTTPStatusCodeFromStatus(s *status.Status) int {
 	case codes.ResourceExhausted:
 		return http.StatusTooManyRequests
 	// Not Retryable
+	case codes.InvalidArgument:
```
Comment (jaronoff97) [inline receiver/otlpreceiver/internal/errors/errors.go:43]: @TylerHelmuth maybe we should this function and its tests to the helper i made last time? https://github.com/open-telemetry/opentelemetry-collector/blob/af33ac5acc08dbeb43426f86b526d2cfd29318ab/internal/httphelper/helper.go#L15C6-L15C33

That way the conversions are in the same place

Comment (TylerHelmuth) [reply receiver/otlpreceiver/internal/errors/errors.go:43]: @jaronoff97 it is all moving (and properly solved by remembering the actual http status code) via https://github.com/open-telemetry/opentelemetry-collector/pull/9041, so we should stick with that PR so the permanent solution to this issue. This PR fixes the bug for now.

Comment (jaronoff97) [reply receiver/otlpreceiver/internal/errors/errors.go:43]: cool! I have no problem with that :D

**Review by codeboten — APPROVED** [review]

**Pull request merged**